### ISSUE TITLE
As a user, I'd like to be able to run the Data Importer without having the Customize Application system permission

### ISSUE THREAD

**Issue by tet3** [SalesforceFoundation/Cumulus#1707]
BDI_DataImport_CTRL.importData() attempts to upsert the custom setting record before starting the data import batch job. If the user does not have the Customize Application permission, then they will get an error that they don't have access:
<img width="1170" alt="screen shot-npspdataimport" src="https://cloud.githubusercontent.com/assets/1319826/9798637/330e189c-57d0-11e5-99fb-617a8734777b.png">

**Comment by cdcarter** [comment]
I'm with Thomas, it seems unnecessarily restrictive to require full access to the setup tree to do data import, especially when the Data Importer is designed to make data importing LESS messy. If there's a desire to restrict it, CRUD on the import object or even a custom permission seem more appropriate than tying it to an un-intuitive permission.

**Comment by kbromer** [comment]
Is this true even if that setting is configured already? 

The only way it should need full CRUD for the setting ot update is if its requiring an update, which i think we DO want to require a higher-level of access to.  I haven't tested it though yet

**Comment by tet3** [comment]
Yes, the [importData method](https://github.com/SalesforceFoundation/Cumulus/blob/dev/src/classes/BDI_DataImport_CTRL.cls#L97) that is called by the "Start Import" button attempts to update the CS record in all but test scenarios:
```
    public PageReference importData() {
    	if(!Test.isRunningTest())
    	   upsert diSettings;
```

**Comment by kbromer** [comment]
Cool, thx for clarifying.

**Comment by davidhabib** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x800000001fssAAA" target="_blank">W-008283</a>

**Comment by BrightStepTracy** [comment]
Can we bump up the priority of this for future sprints?  Thanks!

**Comment by cdcarter** [comment]
Is this two separate issues?

1) That an upsert is attempted on the settings even if they are configured, which is requiring "Configure Application" instead of just throwing an error if there is both NO settings and NO configure application.
2) That there should be a way to permission users in and out of using the import tool.

I don't even know, does the NPSP have any custom permissions in it yet? Is this a place where we want to do that?

**Comment by tet3** [comment]
I think it's really just #1. Access to the tool can be controlled with profile/perm set on the VF page.

**Comment by BrightStepTracy** [comment]
I think as long as the tool itself doesn't require "Configure Application," which is not a standard User system permission, I'm happy.  As to the mechanism, it would be ideal if it could be rolled into a Permission Set inside the NPSP itself, however, this may not be scalable or ideal, since new fields/Objects added to the tool over time would need to be re-added to the Permission Set anyway, it's better to simply allow permissions for the tool itself to be something that can be assigned to a Profile.

**Comment by cdcarter** [comment]
Great point on the VF page perms, @tet3!

**Comment by davidhabib** [comment]
really, all we want to do is have the visualforce page disable the various settings controls if the user doesn't have permissions to modify them.  then the run process code can avoid upserting the settings if they aren't enabled for the user.  very straightforward, except I'm not sure Salesforce provides us the same metadata to know whether we can update a custom setting vs. updating field on an object.  If the field describe call works on settings, this is straight forward.

**Comment by cdcarter** [comment]
Custom settings are describable and have the isUpdatable property on their describe result. This should be pretty easy, I'll look at it today.

**Comment by mrbelvedere** [comment]
Included in beta release version 3.61 (Beta 4) and higher

**Comment by mrbelvedere** [comment]
Included in production release version 3.61
